### PR TITLE
feature: add a paramter labels to CreateSnapshot

### PR DESCRIPTION
**Pull request by firjest**
Signed-off-by: yifang.jy <yifang.jy@alibaba-inc.com>

### Ⅰ. Describe what this PR did
snapshot interface has labels parameter, so add this parameter to pouch CreateSnapshot In some case use can pass labels.

### Ⅱ. Does this pull request fix one issue?

NO

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
NO

### Ⅳ. Describe how to verify it
NO

### CODE REVIEW

File: ctrd/interface.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ type ImageAPIClient interface {
 // SnapshotAPIClient provides access to containerd snapshot features
 type SnapshotAPIClient interface {
 	// CreateSnapshot creates a active snapshot with image's name and id.
-	CreateSnapshot(ctx context.Context, id, ref string) error
+	CreateSnapshot(ctx context.Context, id, ref string, labels *map[string]string) error
```
Comment (Ace-Tang) [inline ctrd/interface.go]: Why we should use `*map[string]string`

File: ctrd/snapshot.go [hunk 2]
```diff
@@ -31,8 +31,13 @@ func (c *Client) CreateSnapshot(ctx context.Context, id, ref string) error {
 		return err
 	}
 
+	var opts []snapshots.Opt
+	if labels != nil && len(*labels) > 0 {
+		opts = append(opts, snapshots.WithLabels(*labels))
```
Comment (Ace-Tang) [inline ctrd/snapshot.go]: How about just `opts := []snapshots.Opt{snapshots.WithLabels(labels)}`, of course, not use `*map` here, see last comment

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -383,7 +383,7 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 
 	snapID := id
 	// create a snapshot with image.
-	if err := mgr.Client.CreateSnapshot(ctx, snapID, config.Image); err != nil {
+	if err := mgr.Client.CreateSnapshot(ctx, snapID, config.Image, nil); err != nil {
```
Comment (Ace-Tang) [inline daemon/mgr/container.go]: Seems we should get somewhere to pass the label, or it will always be nil

Comment (Ace-Tang) [reply daemon/mgr/container.go]: We can add a `snapshot-opt` which keep options for snapshotter

File: ctrd/interface.go (hunk 1 above)
Comment (firjest) [reply ctrd/interface.go]: done

File: ctrd/snapshot.go (hunk 2 above)
Comment (firjest) [reply ctrd/snapshot.go]: done

**Comment by wangforthinker** [comment]
To support this feature, containerd should make some changes like this! https://github.com/alibaba/containerd/pull/5/files#diff-491f07c28eb851b720f11ce0d3dbb4f8R331 
. In current version of v1.0.3, containerd do not pass these opts to snapshotter when call method Prepare.

**Pull request closed without merging**